docs/deployment/pull-backup.rst

```diff
@@ -301,3 +301,108 @@ a backup may be the following command::
 
 This command also automatically removes the socket file after the ``borg
 create`` command is done.
+
+ssh-agent
+=========
+
+In this scenario *borg-server* initiates an SSH connection to *borg-client* and forwards the authentication
+agent connection.
+
+After that, it works similar to the push mode:
+*borg-client* initiates another SSH connection back to *borg-server* using the forwarded authentication agent
+connection to authenticate itself, starts ``borg serve`` and communicates with it.
+
+Using this method requires ssh access of user *borgs* to *borgc@borg-client*, where:
+
+* *borgs* is the user on the server side with read/write access to local borg repository.
+* *borgc* is the user on the client side with read access to files meant to be backed up.
+
+Applying this method for automated backup operations
+----------------------------------------------------
+
+Assume that the borg-client host is untrusted.
+Therefore we do some effort to prevent a hostile user on the borg-client side to do something harmful.
+In case of a fully trusted borg-client the method could be simplified.
+
+Preparing the server side
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Do this once for each client on *borg-server* to allow *borgs* to connect itself on *borg-server* using a
+dedicated ssh key:
+
+::
+
+  borgs@borg-server$ install -m 700 -d ~/.ssh/
+  borgs@borg-server$ ssh-keygen -N '' -t rsa  -f ~/.ssh/borg-client_rsa
+  borgs@borg-server$ { echo -n 'command="borg serve --append-only --restrict-to-path ~/repo",restrict '; cat ~/.ssh/borg-client_rsa.pub; } >> ~/.ssh/authorized_keys
+  borgs@borg-server$ chmod go-w ~/.ssh/authorized_keys
+
+``install -m 700 -d ~/.ssh/``
+
+  Create directory ~/.ssh with correct permissions if it does not exist yet.
+
+``ssh-keygen -N '' -t rsa  -f ~/.ssh/borg-client_rsa``
+
+  Create an ssh key dedicated to communication with borg-client.
```

Comment:
I replaced the places talking about "rsa key" with "ssh key", so this can be updated more easily, e.g. if we want to recommend ed25519 rather.

Reply:
Not sure if we also want to do a similar thing to the key filename?

Reply:
> Not sure if we also want to do a similar thing to the key filename?

Yes, something like ~/.ssh/borg-client_key will be more correctly.

Reply:
ok, sounds good. you will update the PR?

Reply:
Yes.